### PR TITLE
Fixed exception handling for createFile and createDir function calls

### PR DESCRIPTION
**Pull request by y0014984**
By merging this Pull Request every call of createDir or createFile function is embedded in a try/catch-block. Both functions thow errors if the filesystem object already exists. This happens for example if you snyc identical modules multiple times to the same computer. The exceptions are now logged in the *.rpt file.

### CODE REVIEW

File: addons/filesystem/functions/fnc_initFilesystem.sqf [hunk 1]
```diff
@@ -35,26 +35,34 @@ private _filesystemObjects = ("inheritsFrom _x == (configFile >> 'AE3_Filesystem
...
+			[
+				_ptr, 
+				_filesystem, 
+				_path, 
+				(getText (_x >> "content")), 
+				"root", 
+				_owner,
+				_permissions 
+			] call AE3_filesystem_fnc_createFile;
+		} catch { diag_log format ["AE3 exception: %1", _exception]; };
```
Comment (GermanHydrogen) [inline addons/filesystem/functions/fnc_initFilesystem.sqf:50]: To broad catch. Would be better, if other exceptions are still raised

File: addons/filesystem/functions/fnc_initFilesystem.sqf [hunk 2]
```diff
@@ -35,26 +35,34 @@ private _filesystemObjects = ("inheritsFrom _x == (configFile >> 'AE3_Filesystem
...
+		{
+			[
+				_ptr,
+				_filesystem,
+				_path,
+				"root",
+				_owner, 
+				_permissions
+			] call AE3_filesystem_fnc_createDir;
+		} catch { diag_log format ["AE3 exception: %1", _exception]; };
```
Comment (GermanHydrogen) [inline addons/filesystem/functions/fnc_initFilesystem.sqf:65]: See above

File: addons/filesystem/functions/fnc_moduleAddDir.sqf [hunk 3]
```diff
@@ -40,14 +40,19 @@ if(_path isEqualTo "") exitWith {};
...
+		{
+			[
+				[],
+				_filesystem,
+				_path,
+				"root",
+				_owner,
+				_permissions
+			] call AE3_filesystem_fnc_createDir;
+		} catch { diag_log format ["AE3 exception: %1", _exception]; };
```
Comment (GermanHydrogen) [inline addons/filesystem/functions/fnc_moduleAddDir.sqf:55]: See above

File: addons/filesystem/functions/fnc_moduleAddFile.sqf [hunk 4]
```diff
@@ -47,15 +47,21 @@ if(_isFunction) then
...
+			[
+				[],
+				_filesystem,
+				_path,
+				_content,
+				"root",
+				_owner,
+				_permissions
+			] call AE3_filesystem_fnc_createFile;
+		} catch { diag_log format ["AE3 exception: %1", _exception]; };
```
Comment (GermanHydrogen) [inline addons/filesystem/functions/fnc_moduleAddFile.sqf:63]: see above

**Review by GermanHydrogen — CHANGES_REQUESTED** [review]

**Comment by y0014984** [comment]
@GermanHydrogen ,I modified the catch clauses to only catch 'already exists' exceptions, so other exceptions are still thrown. It was a bit complicated because the exception is not a fixed string but contains the name of the filesystem object that already exists. So I needed to regExReplace the dynamic content for example `'home' already exists!` to look like this `'%1' already exists!`. This this the `format` string, that is used to build the dynamic exception. So, I can use this to compare the strings in the `if` clause.

**Review by GermanHydrogen — APPROVED** [review]

**Pull request merged**